oat-sa/tao-core-ui-fe
# Feature/TR-1810/Add option to disable media player preview

Related to: https://oat-sa.atlassian.net/browse/TR-1810

Refactoring of the media player:
- split into smaller and unit testable modules (`players/html5`, `players/youtube`, `support`, `youtubeManager`). However, the main module mostly remains the same (it would need to be rewritten as well)
- apply ES6+
- rework the unit tests to make them work the same both in CLI and in browser (use mocks), and get rid of broken tests
- rework the templates, moving the media related content in each player variant, the media sources being added just in time

New modules from the split:
- `ui/mediaplayer/support`: helper that allows to check the browser support for media and and codecs
- `ui/mediaplayer/youtubeManager`: helper that allows to install the Youtube API and create a Youtube player
- `ui/mediaplayer/players`: list of available media players, index by the type (`audio`, `video`, `youtube`)
- `ui/mediaplayer/players/html5`: HTML5 variant of the media player
- `ui/mediaplayer/players/html5`: Youtube variant of the media player

Added features:
- option `preview`, active by default: allows to preview the media (especially video). When disabled, the player shows a blank placeholder with a play button and the duration is unknown.
![image](https://user-images.githubusercontent.com/1500098/130093800-fdf8c930-e9c7-4204-9916-c2b9a58c7c2e.png)
- option `debug`, disabled by default: allows to add some verbosity in the console about the player's behavior (essentially at the level of the player implementation).
- added common config for Prettier and ESLint

How to test:
- the unit tests must pass (`mediaplayer/*`)
- `npm run test mediaplayer`
- `npm run test:keepAlive`, then open the browser at the given location, and browse to each unit test (see list below)
- additionally, check the companion PR for more info

Impacted/Added unit tests:
- `test/mediaplayer/test.html`
- `test/mediaplayer/players/test.html`
- `test/mediaplayer/players/html5/test.html`
- `test/mediaplayer/players/youtube/test.html`
- `test/mediaplayer/support/test.html`
- `test/mediaplayer/youtubeManager/test.html`
